veles/secrets/postmanapikey/validator_test.go

```diff
@@ -0,0 +1,462 @@
+		if r.Method != http.MethodGet || r.URL.Path != expectedPath {
+			t.Errorf("unexpected request: %s %s, expected: GET %s", r.Method, r.URL.Path, expectedPath)
+			http.Error(w, "not found", http.StatusNotFound)
+			return
+		}
+
+		// Check access_key query parameter
+		if expectedKey != "" {
+			accessKey := r.URL.Query().Get("access_key")
+			if accessKey != expectedKey {
+				t.Errorf("expected access_key query parameter to be %s, got: %s", expectedKey, accessKey)
+			}
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(statusCode)
+		if body != nil {
+			_ = json.NewEncoder(w).Encode(body)
+		}
+	}))
+}
+
+func TestValidatorAPI(t *testing.T) {
+	cases := []struct {
+		name        string
+		statusCode  int
+		body        any
+		want        veles.ValidationStatus
+		expectError bool
+	}{
+		{
+			name:       "valid_key",
+			statusCode: http.StatusOK,
+			body: map[string]any{
+				"user": map[string]any{
+					"id":   12345,
+					"name": "Test User",
+				},
+			},
+			want: veles.ValidationValid,
+		},
+		{
+			name:       "invalid_key_unauthorized",
+			statusCode: http.StatusUnauthorized,
+			body: map[string]any{
+				"error": map[string]any{
+					"name":    "AuthenticationError",
+					"message": "Invalid API Key. Every request requires a valid API Key to be sent.",
+				},
+			},
+			want: veles.ValidationInvalid,
+		},
+		{
+			name:        "server_error",
+			statusCode:  http.StatusInternalServerError,
+			body:        nil,
+			want:        veles.ValidationFailed,
+			expectError: true,
+		},
+		{
+			name:        "forbidden_error",
+			statusCode:  http.StatusForbidden,
+			body:        nil,
+			want:        veles.ValidationFailed,
+			expectError: true,
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			// Create mock server
+			server := mockAPIServer(t, validatorTestAPIKey, tc.statusCode, tc.body)
+			defer server.Close()
+
+			// Create client with custom transport
+			client := &http.Client{
+				Transport: &mockTransport{testServer: server},
+			}
+
+			// Create validator with mock client
+			validator := postmanapikey.NewAPIValidator(
+				postmanapikey.WithClientAPI(client),
+			)
+
+			// Create test key
+			key := postmanapikey.PostmanAPIKey{Key: validatorTestAPIKey}
+
+			// Test validation
+			got, err := validator.Validate(context.Background(), key)
+
+			// Check error expectation
+			if tc.expectError {
+				if err == nil {
+					t.Errorf("Validate() expected error, got nil")
+				}
+			} else {
+				if err != nil {
+					t.Errorf("Validate() unexpected error: %v", err)
+				}
+			}
+
+			// Check validation status
+			if got != tc.want {
+				t.Errorf("Validate() = %v, want %v", got, tc.want)
+			}
+		})
+	}
+}
+
+func TestValidatorAPI_ContextCancellation(t *testing.T) {
+	// Create a server that delays response
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		time.Sleep(100 * time.Millisecond)
```

Comment:
It should be possible to use contexts that cancel immediately with so that we can avoid having to call time.Sleep. Example: https://github.com/google/osv-scalibr/blob/6f83d8c293d88786a5f717166ce719f2eea086f6/veles/secrets/gcpsak/validator_test.go#L189